okhttp-logging-interceptor/src/main/kotlin/okhttp3/logging/HttpLoggingInterceptor.kt

```diff
@@ -312,6 +322,32 @@ class HttpLoggingInterceptor
       return response
     }
 
+    private fun sanitizeUrl(url: HttpUrl): String {
```

Comment:
Actually, I think this needs rework.

queryParameterNames is a unique Set.

Maybe something like

```
    private fun sanitizeUrl(url: HttpUrl): String {
      if (queryParamsNameToRedact.isEmpty() || url.querySize == 0) {
        return url.toString()
      }

      val newUrl = url.newBuilder().query(null).apply {
        for (i in 0 until url.querySize) {
          val parameterName = url.queryParameterName(i)
          val newValue = if (parameterName in queryParamsNameToRedact) "██" else url.queryParameterValue(i)

          addEncodedQueryParameter(parameterName, newValue)
        }
      }.toString()
    }
```

Reply:
Thanks for your suggestion; this one is definitely better than my string concatenation solution.